joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/parser/ParserAst.scala

```diff
@@ -0,0 +1,221 @@
+    body: ParserRuleContext
+  ) extends ParserNode(ctx)
+
+  final case class MandatoryParameter(ctx: ParserRuleContext) extends ParserNode(ctx)
+
+  final case class OptionalParameter(
+    ctx: ParserRuleContext,
+    name: ParserRuleContext,
+    defaultExpression: ParserRuleContext
+  ) extends ParserNode(ctx)
+
+  final case class ArrayParameter(ctx: ParserRuleContext, name: Option[String]) extends ParserNode(ctx)
+
+  final case class HashParameter(ctx: ParserRuleContext, name: Option[String]) extends ParserNode(ctx)
+
+  final case class ProcParameter(ctx: ParserRuleContext, name: ParserRuleContext) extends ParserNode(ctx)
+
+  final case class SingleAssignment(ctx: ParserRuleContext, lhs: ParserRuleContext, op: String, rhs: ParserRuleContext)
+      extends ParserNode(ctx)
+
+  final case class AttributeAssignment(
+    ctx: ParserRuleContext,
+    target: ParserRuleContext,
+    op: String,
+    attributeName: String,
+    rhs: ParserRuleContext
+  ) extends ParserNode(ctx)
+
+  final case class WhileExpression(ctx: ParserRuleContext, condition: ParserRuleContext, body: ParserRuleContext)
+      extends ParserNode(ctx)
+
+  final case class UntilExpression(ctx: ParserRuleContext, condition: ParserRuleContext, body: ParserRuleContext)
+      extends ParserNode(ctx)
+
+  final case class IfExpression(
+    ctx: ParserRuleContext,
+    condition: ParserRuleContext,
+    thenClause: ParserRuleContext,
+    elsifClauses: List[ParserRuleContext],
+    elseClause: Option[ParserRuleContext]
+  ) extends ParserNode(ctx)
+
+  final case class ElsIfClause(ctx: ParserRuleContext, condition: ParserRuleContext, thenClause: ParserRuleContext)
+      extends ParserNode(ctx)
+
+  final case class ElseClause(ctx: ParserRuleContext, thenClause: ParserRuleContext) extends ParserNode(ctx)
+
+  final case class UnlessExpression(
+    ctx: ParserRuleContext,
+    condition: ParserRuleContext,
+    trueBranch: ParserRuleContext,
+    falseBranch: Option[ParserRuleContext]
+  ) extends ParserNode(ctx)
+
+  final case class ConditionalExpression(
+    ctx: ParserRuleContext,
+    condition: ParserRuleContext,
+    trueBranch: ParserRuleContext,
+    falseBranch: ParserRuleContext
+  ) extends ParserNode(ctx)
+
+  final case class ReturnExpression(ctx: ParserRuleContext, expressions: List[ParserRuleContext])
+      extends ParserNode(ctx)
+
+  /** Represents an unqualified identifier e.g. `X`, `x`, `@x`, `@@x`, `$x`, `$<`, etc. */
+  final case class SimpleIdentifier(ctx: ParserRuleContext, typeFullName: Option[String] = None) extends ParserNode(ctx)
+
+  final case class SelfIdentifier(ctx: ParserRuleContext) extends ParserNode(ctx)
+
+  /** Represents a non-interpolated literal. */
+  final case class StaticLiteral(ctx: ParserRuleContext, typeFullName: String) extends ParserNode(ctx) {
+    def isSymbol: Boolean = ctx.getText.startsWith(":")
+    def isString: Boolean = ctx.getText.startsWith("\"")
+
+    // TODO: ugly as hell.
```

Comment:
Lol, the other solution I can think of is pattern matching, i.e.,:
```scala
  def innerText: String = ctx.getText.toList match {
    case ':' :: '\'' :: tail => tail.dropRight(1)
    case ':' :: tail         => tail
    case '\'' :: tail        => tail.dropRight(1)
    case _                   => ctx.getText
  }
```

Reply:
it's also possible to pattern-match on s-strings, i.e.
```scala
case s":'$content'" => content
case s":$symbol" => symbol
case s"'$content'" => content
case _ => ctx.getText
```

Reply:
I should probably pattern match on `ctx` instead, but that would require that I rearrange some ANTLR rules beforehand and it just didn't seem that high priority at the moment. But yeah, this hurts to look at. Thanks for the suggestion as well :)

Reply:
> it's also possible to pattern-match on s-strings

Cool! I didn't know that. Thanks for mentioning it!